### PR TITLE
exception handling docs

### PR DESCRIPTION
**Pull request by juliagsy**

### CODE REVIEW

File: docs/partial_source/deep_dive/18_exception_handling.rst
Comment (djl11) [inline docs/partial_source/deep_dive/18_exception_handling.rst:47]: The :code:`IvyError` class unifies these special cases.

**Review by djl11 — CHANGES_REQUESTED** [review]
very minor changes, and all just about the writing style. I think it's better to use language less tied to us as creators. For example avoiding phrases like "we created <class_name> because..." and instead having phrases like "<class_name> does this.", if that makes sense :) Once modified, feel free to merge without requesting another review!

**Comment by juliagsy** [comment]
Noted! Will take note in future writing! Thanks for the review!

**Pull request merged**
